### ISSUE TITLE
Ubuntu 18.04 installation does not work

### ISSUE THREAD

**Issue by dalibor983** [powerline/fonts#281]
I do not see any of this fonts after installing it with
```
sudo apt-get install fonts-powerline
sudo fc-cache -f -v
```

I tried this also [fontconfig](https://github.com/powerline/fonts#fontconfig) suggestion but without success.

**Comment by Darkbish** [comment]
Yes, apt package does not work, I clone this repository and install by `install.sh` works.

**Comment by likarum** [comment]
Debian 9.4 same problem (and same solution)

**Comment by SConaway** [comment]
Same.

**Comment by pandasauce** [comment]
18.04 LTS, neither option works.

**Comment by cieplik206** [comment]
Same, running install.sh also does not fix font on shell with ohmyzsh, any ideas ?

**Comment by lazarocastro** [comment]
Try to use that in your terminal:

```
git clone https://github.com/powerline/fonts.git --depth=1
cd fonts
./install.sh
cd ..
rm -rf fonts
```
https://github.com/powerline/fonts

**Comment by AntoniosHadji** [comment]
18.04 - I have the glyphs showing in gvim with extra garbage characters. In terminal, the glyghs do not show at all.  I have tried many variations on these instructions.  What is the best way to debug this issue?
In this image, the top is terminal vim in gnome terminal with font set to patched Ubuntu Mono. The lower section is gvim with same font.  
![screenshot from 2018-11-27 20-31-25](https://user-images.githubusercontent.com/1932568/49125018-7fe72c80-f28c-11e8-9b81-14da6a4d32e6.png)

**Comment by toxicmender** [comment]
Just in case:
1. clone this repo
2. run ```./install.sh``` from the repo
3. add the files [10](https://github.com/powerline/powerline/raw/develop/font/10-powerline-symbols.conf) & [50](https://github.com/powerline/fonts/blob/master/fontconfig/50-enable-terminess-powerline.conf) to fontconfig/conf.d/ directory 
    **Note:** _fc-cache -vf in Ubuntu Disco checks in ~/.fontconfig/conf.d, it could be ~/.config/fontconfig/conf.d/ otherwise_
4. Now open your terminal emulation application and go to preferences->profile and change the font to the patched font. For example: Mine was ```FiraMono Regular```, changed to ```FiraMono for powerline```.
5. Save and the exit preferences, you should see the new font now

**Comment by Jooeeee** [comment]
Maybe, it doesn't have the permission. 
try this:
 install font, then add permission;
 `chmod -R 777 ~/.local/share/fonts`

**Comment by apjanke** [comment]
The Debian `fonts-powerline` package is not expected to work this way. That `fonts-powerline` package contains only the "Powerline Symbols" font, which is a small font containing only the special Powerline symbol characters, and no normal characters (like "A", "B", "C", etc.). On its own, the Powerline Symbols font is only useful in applications which support "font fallback", where you can specify multiple fonts, and if the application doesn't find a glyph for a particular character in the first font, it goes and looks in the second font, and so on. Lots of apps don't support this.

What y'all want are the "Powerline-patched fonts", which are a collection of various open-source fonts which have had the plain Powerline Symbols font merged in to them, so that you can use them in applications which support only a single font selection.

Maybe someone should send in a request to Debian for a `fonts-powerline-patched` package, that contains the powerline-patched fonts in this repo? Could make life easier for users.

**Comment by leijerry888** [comment]
> The Debian `fonts-powerline` package is not expected to work this way. That `fonts-powerline` package contains only the "Powerline Symbols" font, which is a small font containing only the special Powerline symbol characters, and no normal characters (like "A", "B", "C", etc.). On its own, the Powerline Symbols font is only useful in applications which support "font fallback", where you can specify multiple fonts, and if the application doesn't find a glyph for a particular character in the first font, it goes and looks in the second font, and so on. Lots of apps don't support this.
> 
> What y'all want are the "Powerline-patched fonts", which are a collection of various open-source fonts which have had the plain Powerline Symbols font merged in to them, so that you can use them in applications which support only a single font selection.
> 
> Maybe someone should send in a request to Debian for a `fonts-powerline-patched` package, that contains the powerline-patched fonts in this repo? Could make life easier for users.

This is exactly what happened to me! The symbols did help me to have a good ohmyzsh in ubuntu terminal, but not in vscode terminal. Now I will need a xxx for powerline, i.e. powerline patched font. This is on ubuntu 20.04 beta

**Comment by entrptaher** [comment]
> Just in case:
> 
> 1. clone this repo
> 2. run `./install.sh` from the repo
> 3. add the files [10](https://github.com/powerline/powerline/raw/develop/font/10-powerline-symbols.conf) & [50](https://github.com/powerline/fonts/blob/master/fontconfig/50-enable-terminess-powerline.conf) to fontconfig/conf.d/ directory
>    **Note:** _fc-cache -vf in Ubuntu Disco checks in ~/.fontconfig/conf.d, it could be ~/.config/fontconfig/conf.d/ otherwise_
> 4. Now open your terminal emulation application and go to preferences->profile and change the font to the patched font. For example: Mine was `FiraMono Regular`, changed to `FiraMono for powerline`.
> 5. Save and the exit preferences, you should see the new font now

I did have to change the fonts, no other ways were working because it was simply using a different font. Both for Terminator and gnome-terminal.

![](https://i.imgur.com/nVF3EV0.png)

**Comment by the-lambda-way** [comment]
> This is exactly what happened to me! The symbols did help me to have a good ohmyzsh in ubuntu terminal, but not in vscode terminal. Now I will need a xxx for powerline, i.e. powerline patched font. This is on ubuntu 20.04 beta

I had the same issue with VSCode on Ubuntu, which led me to the thread. You can add PowerlineSymbols as a fallback font to the terminal simply by adding it to the "terminal.integrated.fontFamily" setting (ex: `my_font, PowerlineSymbols`). However, I found the symbols to be slightly misaligned to my font (DejaVu Sans Mono), so I ended up using the patched version anyway.

**Comment by Mithrandir2k18** [comment]
> > This is exactly what happened to me! The symbols did help me to have a good ohmyzsh in ubuntu terminal, but not in vscode terminal. Now I will need a xxx for powerline, i.e. powerline patched font. This is on ubuntu 20.04 beta
> 
> I had the same issue with VSCode on Ubuntu, which led me to the thread. You can add PowerlineSymbols as a fallback font to the terminal simply by adding it to the "terminal.integrated.fontFamily" setting (ex: `my_font, PowerlineSymbols`). However, I found the symbols to be slightly misaligned to my font (DejaVu Sans Mono), so I ended up using the patched version anyway.

Same issue on Ubuntu 20.04

**Comment by LinkPhoenix** [comment]
### With Pop!_OS 21.10

I create **powerline** folder in **/usr/share/fonts/** en Reset font cache and all is work !

```
$ git clone https://github.com/powerline/fonts.git
$ sudo mkdir /usr/share/fonts/powerline
$ sudo cp -r ~/fonts/* /usr/share/fonts
$ fc-cache -f /usr/share/fonts
```
Exit all terminals and launch one

With the **./install.sh** the script install fonts in **/.local/share/fonts** but didn't work

**Comment by NTD815** [comment]
> ### With Pop!_OS 21.10
> I create **powerline** folder in **/usr/share/fonts/** en Reset font cache and all is work !
> 
> ```
> $ git clone https://github.com/powerline/fonts.git
> $ sudo mkdir /usr/share/fonts/powerline
> $ sudo cp -r ~/fonts/* /usr/share/fonts
> $ fc-cache -f /usr/share/fonts
> ```
> 
> Exit all terminals and launch one
> 
> With the **./install.sh** the script install fonts in **/.local/share/fonts** but didn't work

Thanks a lot! Working now ;)
